src/test/objectstore/allocsim/ops_parser.cc

```diff
@@ -0,0 +1,236 @@
+#include "ops_parser.h"
+#include <cassert>
+#include <fmt/format.h>
+#include <iostream>
+#include <thread>
+#include <map>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+using namespace std;
+
+int64_t timestamp_parser2(const char* t) {
+  // expected format
+  // 2024-05-10 12:06:24.792232+00:00
+  // 0123456789012345678------------
+  static constexpr uint32_t time_len = sizeof("2024-05-10 12:06:24");
+  thread_local char previous_str[time_len]("0000-00-00 00:00:00");
+  thread_local uint64_t previous_time = 0;
+  int usec = atoi(t + 20);
+  if (usec < 0 || usec > 999999) return 0;
+  if (strncmp(t, previous_str, time_len) == 0) {
+    return previous_time * 1000000 + usec;
+  }
+  struct tm a_tm;
```

Comment:
isn't this whole parse code more or less equivalent to
  sys_time<nanoseconds> tp;
  in >> std::chrono::parse("%F %T", tp);
(assuming 'in' contains the text)?

See https://godbolt.org/z/14e5Ybrsq for a full example you can change to fit your needs